src/status_test.rs

```diff
@@ -661,3 +661,341 @@ mod is_disconnected {
+        ];
+
+        let mut new_state = get_new_state_msg();
+        new_state.received = Utc::now() - Duration::minutes(5);
+        let mut latest_new_state = get_new_state_msg();
+        latest_new_state.received = Utc::now() - Duration::zero();
+        let approve_state = get_approve_state_msg(true);
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(leader_heartbeats),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(follower_heartbeats),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_rejected_state(&messages, &latest_new_state),
+            false,
+            "Last approved newState is older than latest newstate but NOT older than a minute"
+        )
+    }
+    #[test]
+    fn approved_and_latest_new_state_are_the_same() {
+        let channel = DUMMY_CHANNEL.clone();
+        let leader_heartbeats = vec![
+            get_heartbeat_msg(Duration::zero(), channel.spec.validators.leader().id),
+            get_heartbeat_msg(Duration::zero(), channel.spec.validators.follower().id),
+        ];
+        let follower_heartbeats = vec![
+            get_heartbeat_msg(Duration::zero(), channel.spec.validators.leader().id),
+            get_heartbeat_msg(Duration::zero(), channel.spec.validators.follower().id),
+        ];
+
+        let five_min_ago = Utc::now() - Duration::minutes(5);
+        let mut new_state = get_new_state_msg();
+        new_state.received = five_min_ago;
+        let mut latest_new_state = get_new_state_msg();
+        latest_new_state.received = five_min_ago;
+        let approve_state = get_approve_state_msg(true);
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: Some(new_state),
+                    approve_state: None,
+                }),
+                heartbeats: Some(leader_heartbeats),
+            },
+            follower: LastApprovedResponse {
+                last_approved: Some(LastApproved {
+                    new_state: None,
+                    approve_state: Some(approve_state),
+                }),
+                heartbeats: Some(follower_heartbeats),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
+            is_rejected_state(&messages, &latest_new_state),
+            false,
+            "Last approved new state and latest new state are the same message and it is older than a minute"
+        )
+    }
+}
```

Comment:
Missing test:
> Last approved newState is neither older than the latest nor older than a minute returns false

Reply:
I originally had this test because it's in the market, however if the last approved newState is not older than the latest newState it means that the latest one is the last approved one. That case is already covered in the test above.

Reply:
Added